### ISSUE TITLE
DataStore observeQuery is slow unless AWSAPIPlugin is added

### ISSUE THREAD

**Issue by chernushevich** [aws-amplify/amplify-swift#3100]
### Describe the bug

DataStore observeQuery takes anywhere between 1 and 3 seconds to return new snapshot. Unless AWSAPIPlugin is added, then it works instantly

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Follow the tutorial at https://docs.amplify.aws/start/q/integration/ios/, do NOT add AWSAPIPlugin
2. Add Amplify.DataStore.observeQuery to monitor Todos
3. Add a new Todo with Amplify.DataStore.save
4. New items take anywhere between 1 and 3 seconds to appear, visible delays
5. Add and configure AWSAPIPlugin
6. The exact same code now adds new items instantly
```

### Expected behavior

Amplify.DataStore.observeQuery returns new items instantly in offline mode

### Amplify Framework Version

2.15.0

### Amplify Categories

API, DataStore

### Dependency manager

Swift PM

### Swift version

5.9

### CLI version

12.2.0

### Xcode version

15.0 Beta 4 (15A5195m)

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

No

### Regression additional context

_No response_

### Platforms

_No response_

### OS Version

macOS Ventura 13.4.1

### Device

iPhone Xr

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by chernushevich** [comment]
Calling `Amplify.DataStore.stop()` right before Amplify.DataStore.save fixes the issue.

**Comment by atierian** [comment]
We've opened a PR to address this.
- https://github.com/aws-amplify/amplify-swift/pull/3214

We'll follow up to let you know once it's released.
Thanks for reporting the issue @chernushevich!

**Comment by atierian** [comment]
The fix for this was released in [2.25.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.25.1). Thanks again for reporting this and for your patience!
